src/lib/sw.js

```diff
@@ -1,31 +1,88 @@
+const normalizeIndexCacheKeyPlugin = {
+  cacheKeyWillBeUsed({request, mode}) {
+    // Take advantage of Workbox's built-in handling of .../index.html routes and ensure that its
+    // cache keys always include it. (e.g., requests for foo/ will load foo/index.html: but
+    // foo/index.html will not load foo/).
+    if (request.url.endsWith("/")) {
+      return request.url + "index.html";
+    }
+    return request;
+  },
+};
+
 import manifest from "cache-manifest";
 
 importScripts(
   "https://storage.googleapis.com/workbox-cdn/releases/4.3.1/workbox-sw.js",
 );
 
-console.log("Got cache manifest", manifest);
+workbox.precaching.precacheAndRoute(manifest);
 
-workbox.precaching.precache(["/offline/"]);
+/**
+ * Match "/foo-bar/ "and "/foo-bar/as/many/of-these-as-you-like/" (with optional trailing
+ * "index.html"), normal page nodes for web.dev.
+ */
+const contentPageRe = new RegExp("/([\\w-]+/)*(|index.html)$");
+
+/**
+ * Match "/foo-bar" and "/foo-bar/as-many/but/no/trailing/slash" (but not "/foo/bar/index.html").
+ */
+const untrailedContentPageRe = new RegExp("(/[\\w-]+)+$");
+
+/**
+ * Send normal nodes to cache first.
+ */
+workbox.routing.registerRoute(
+  contentPageRe,
+  new workbox.strategies.StaleWhileRevalidate({
+    plugins: [normalizeIndexCacheKeyPlugin],
+  }),
+);
 
 /**
- * Match /foo-bar/ and "/foo-bar/as/many/of-these-as-you-like/".
- *
- * TODO(samthor): Handle /index.html suffix via cacheKey and the Express-generated 302 from "/foo"
- * to "/foo/".
- *
- * TODO(samthor): Workbox is agressive and will also match "other-domain.com/foo/", include domain
- * restriction.
+ * Cache images that aren't included in the original manifest, such as author profiles.
  */
 workbox.routing.registerRoute(
-  new RegExp("/([\\w-]+/)*$"),
+  new RegExp("/images/.*"),
   new workbox.strategies.StaleWhileRevalidate(),
 );
 
-workbox.routing.setCatchHandler(({event}) => {
+/**
+ * Untrailed requests are network-only, but with a fallback to redirecting to the same page with a
+ * trailing slash.
+ */
+self.addEventListener("fetch", (event) => {
```

Comment:
Does this mean if we leave a `/` off of a link it will try to fetch from the network first?

Reply:
@samthor This part is a bit buried in the docs, but would https://developers.google.com/web/tools/workbox/reference-docs/latest/workbox.precaching.html#~urlManipulation help?

It's designed to let you use the built-in `workbox-precaching` `fetch` handler while providing a custom list of alternative cache keys to check.